diff -ur openssl-0.9.8h.orig/crypto/o_str.c openssl-0.9.8h/crypto/o_str.c
--- openssl-0.9.8h.orig/crypto/o_str.c	2008-03-29 22:22:49.000000000 +0900
+++ openssl-0.9.8h/crypto/o_str.c	2008-05-31 20:24:48.474540000 +0900
@@ -60,7 +60,7 @@
 #include <e_os.h>
 #include "o_str.h"
 
-#if !defined(OPENSSL_IMPLEMENTS_strncasecmp) && !defined(OPENSSL_SYSNAME_WIN32)
+#if !defined(OPENSSL_IMPLEMENTS_strncasecmp) && !defined(OPENSSL_SYSNAME_WIN32) && !defined(OPENSSL_SYSNAME_WINCE)
 # include <strings.h>
 #endif
 
