diff -uNr wcecompat.orig/include/time.h wcecompat/include/time.h
--- wcecompat.orig/include/time.h	2002-11-16 02:45:44.000000000 +0900
+++ wcecompat/include/time.h	2006-05-06 12:06:33.803661000 +0900
@@ -31,12 +31,16 @@
 #include <stdlib.h>		// time_t defined there
 //typedef int time_t;
 
+#ifndef _CLOCK_T_DEFINED
 typedef unsigned long clock_t;
+#define _CLOCK_T_DEFINED
+#endif
 #define CLOCKS_PER_SEC (1000)
 
 time_t time(time_t* t);
 clock_t __cdecl clock(void);
 
+#ifndef _TM_DEFINED
 struct tm
 {
 	int tm_sec;
@@ -49,6 +53,8 @@
 	int tm_yday;
 	int tm_isdst;
 };
+#define _TM_DEFINED
+#endif
 
 struct tm* localtime(const time_t* clock);
 struct tm * __cdecl gmtime(const time_t *clock);
diff -uNr wcecompat.orig/src/wce211_string.c wcecompat/src/wce211_string.c
--- wcecompat.orig/src/wce211_string.c	2002-12-03 02:20:08.000000000 +0900
+++ wcecompat/src/wce211_string.c	2006-05-06 12:06:33.803661000 +0900
@@ -21,6 +21,7 @@
 
 #include <string.h>
 #include <stdlib.h>
+#include <windows.h>
 
 
 _CRTIMP int __cdecl _stricmp(const char* s1, const char* s2)
@@ -72,3 +73,46 @@
 	strcpy(new_s, s);
 	return new_s;
 }
+
+_CRTIMP size_t  __cdecl strspn(const char * s1, const char * s2)
+{
+	const char* p1;
+	const char* p2;
+	
+	for (p1 = s1; *p1; ++p1) {
+		for (p2 = s2; *p2 && *p1 != *p2; ++p2)
+			;
+		if (!*p2)
+			return p1 - s1;
+	}
+	
+	return p1 - s1;
+}
+
+_CRTIMP long   __cdecl strtol(const char * s, char ** endptr, int base)
+{
+	size_t len = strlen(s);
+	wchar_t* ws = _alloca(len + 1);
+	wchar_t* wendptr = 0;
+	long n = 0;
+	
+	MultiByteToWideChar(CP_ACP, 0, s, len, ws, len);
+	n = wcstol(ws, &wendptr, base);
+	if (endptr)
+		*endptr = wendptr ? (char*)s + (wendptr - ws) : 0;
+	return n;
+}
+
+_CRTIMP unsigned long __cdecl strtoul(const char * s, char ** endptr, int base)
+{
+	size_t len = strlen(s);
+	wchar_t* ws = _alloca(len + 1);
+	wchar_t* wendptr = 0;
+	unsigned long n = 0;
+	
+	MultiByteToWideChar(CP_ACP, 0, s, len, ws, len);
+	n = wcstoul(ws, &wendptr, base);
+	if (endptr)
+		*endptr = wendptr ? (char*)s + (wendptr - ws) : 0;
+	return n;
+}
diff -uNr wcecompat.orig/wcedefs.mak wcecompat/wcedefs.mak
--- wcecompat.orig/wcedefs.mak	2005-08-11 10:31:42.000000000 +0900
+++ wcecompat/wcedefs.mak	2006-05-06 22:09:57.905904000 +0900
@@ -54,7 +54,7 @@
 WCETARGETCPUDEFS=-DSH3 -D_SH3_ -DSHx
 !ELSEIF "$(TARGETCPU)"=="SH4"
 WCETARGETCPU=$(TARGETCPU)
-WCETARGETCPUDEFS=-DSH4 -D_SH4_ -DSHx
+WCETARGETCPUDEFS=-DSH4 -D_SH4_ -DSHx -Qsh4
 !ELSEIF "$(TARGETCPU)"=="R4100" || "$(TARGETCPU)"=="R4111" || "$(TARGETCPU)"=="R4300"
 WCETARGETCPU=$(TARGETCPU)
 WCETARGETCPUDEFS=-DMIPS -D_MIPS_ -DMIPS_R4000
