diff -uNr qdbm-1.8.33.orig/VCmakefile qdbm-1.8.33/VCmakefile
--- qdbm-1.8.33.orig/VCmakefile	2005-06-29 09:42:45.000000000 +0900
+++ qdbm-1.8.33/VCmakefile	2005-10-15 21:26:48.019131200 +0900
@@ -18,7 +18,7 @@
 # Debug configuration
 !IF "$(CFG)" == "Debug" || "$(CFG)" == "debug"
 !MESSAGE Build using debug configuration
-BASE_FLAGS = /MLd /W3 /ZI /Od /FD /GZ /Fo"$(OUTDIR)\\" /Fd"$(OUTDIR)\\"
+BASE_FLAGS = /MLd /W3 /Zi /Od /FD /Fo"$(OUTDIR)\\" /Fd"$(OUTDIR)\\"
 BASE_DEFS = -D_DEBUG -D__DEBUG__
 OUTDIR = .\Debug
 LIB_APPEND = _d
@@ -27,7 +27,7 @@
 # Threaded debug configuration
 !ELSEIF "$(CFG)" == "TDebug" || "$(CFG)" == "tdebug"
 !MESSAGE Build using threaded debug configuration
-BASE_FLAGS = /MDd /W3 /ZI /Od /FD /GZ /Fo"$(OUTDIR)\\" /Fd"$(OUTDIR)\\"
+BASE_FLAGS = /MDd /W3 /Zi /Od /FD /Fo"$(OUTDIR)\\" /Fd"$(OUTDIR)\\"
 BASE_DEFS = -D_DEBUG -D__DEBUG__
 OUTDIR = .\TDebug
 LIB_APPEND = _td
@@ -45,7 +45,7 @@
 # Static threaded debug configuration
 !ELSEIF "$(CFG)" == "STDebug" || "$(CFG)" == "stdebug"
 !MESSAGE Build using static threaded debug configuration
-BASE_FLAGS = /MTd /W3 /ZI /Od /FD /GZ /Fo"$(OUTDIR)\\" /Fd"$(OUTDIR)\\"
+BASE_FLAGS = /MTd /W3 /Zi /Od /FD /Fo"$(OUTDIR)\\" /Fd"$(OUTDIR)\\"
 BASE_DEFS = -D_DEBUG -D__DEBUG__
 OUTDIR = .\STDebug
 LIB_APPEND = _std
@@ -72,9 +72,9 @@
 !ENDIF
 
 # Building binaries
-CLFLAGS = /I "$(VCPATH)\Include" /I "$(VCPATH)\PlatformSDK\Include" /I "." /nologo $(YOUR_CLFLAGS) $(BASE_FLAGS) $(BASE_DEFS)
-LIBFLAGS = /libpath:"$(VCPATH)\lib" /libpath:"$(VCPATH)\PlatformSDK\Lib" /libpath:"." /nologo $(YOUR_LIBFLAGS)
-LINKFLAGS = /libpath:"$(VCPATH)\lib" /libpath:"$(VCPATH)\PlatformSDK\Lib" /libpath:"." /nologo $(YOUR_LINKFLAGS)
+CLFLAGS = /I . /nologo $(YOUR_CLFLAGS) $(BASE_FLAGS) $(BASE_DEFS)
+LIBFLAGS = /nologo $(YOUR_LIBFLAGS)
+LINKFLAGS = /nologo $(YOUR_LINKFLAGS) bufferoverflowu.lib
 
 # Targets
 MYLIBS = qdbm$(LIB_APPEND).lib
